homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,216 @@
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.CLIMATE, async_add_entities)
+
+
+class MatterClimate(MatterEntity, ClimateEntity):
+    """Representation of a Matter climate entity."""
+
+    features: int | None = None
+
+    _attr_temperature_unit: str = UnitOfTemperature.CELSIUS
+    _attr_supported_features: ClimateEntityFeature = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        value = int(
+            self.get_matter_attribute_value(clusters.Thermostat.Attributes.SystemMode)
+        )
+        SystemMode = clusters.Thermostat.Enums.ThermostatSystemMode
+
+        match value:
+            case SystemMode.kAuto:
+                return HVACMode.AUTO
+            case SystemMode.kDry:
+                return HVACMode.DRY
+            case SystemMode.kFanOnly:
+                return HVACMode.FAN_ONLY
+            case SystemMode.kCool | SystemMode.kPrecooling:
+                return HVACMode.COOL
+            case SystemMode.kHeat | SystemMode.kEmergencyHeat:
+                return HVACMode.HEAT
+            case _:
+                return HVACMode.OFF
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """Return the list of (currently) available hvac operation modes."""
+        value = int(
+            self.get_matter_attribute_value(
+                clusters.Thermostat.Attributes.ControlSequenceOfOperation
+            )
+        )
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingAndHeating,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingAndHeatingWithReheat,
+        ):
+            return [HVACMode.HEAT, HVACMode.COOL]
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingOnly,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingWithReheat,
+        ):
+            return [HVACMode.COOL]
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kHeatingOnly,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kHeatingWithReheat,
+        ):
+            return [HVACMode.HEAT]
+        return [HVACMode.OFF]
+
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Return the current running hvac operation (if supported)."""
+        if running_state := self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.ThermostatRunningState
+        ):
+            if running_state == 0:
+                return HVACAction.HEATING
+            if running_state == 1:
+                return HVACAction.COOLING
+            if running_state == 2:
+                return HVACAction.FAN
+        if running_mode := self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.ThermostatRunningMode
```

Comment:
What is the meaning of running mode compared to running state? Does it really mean the same thing?

Reply:
the specs are very unclear and like interpretable on multiple ways. Most likely the ThermostatRunningState is comparable to what we call hvac_action

Reply:
Yes, I haven't had the chance to fully understand this part yet

Reply:
I figured it out, as @marcelveldt said, ThermostatRunningState is similar to hvac action. ThermostatRunningMode is used to indicate whether the thermostat is in cool mode or heat mode when the SystemMode is set to Auto. However, when you are in this mode, you may not necessarily be cooling or heating, just like when your SystemMode is set to cool or heat mode

Reply:
However, in the matter spec, ThermostatRunningState is an optional attribute...

Reply:
In ha, is hvac_action mandatory or still optional? I see the default value is None

Reply:
optional